NeuroID/DataStore.swift

```diff
@@ -64,11 +65,22 @@ public enum DataStore {
 
     static func insertCleanedEvent(event: NIDEvent, storeType: String) {
         if storeType == "queue" {
+            // If queue has more than 2000 events, send a queue full event and return
+            if (DataStore.queuedEvents.count >= 2000 && DataStore.events.last?.type != NIDEventName.bufferFull.rawValue) {
+                var fullEvent = NIDEvent.init(type: NIDEventName.bufferFull)
+                DataStore.queuedEvents.append(fullEvent)
+                return
+            }
```

Comment:
While it is possible, I don't see us ever getting to 2000 (or above 2) events in the queued events because we only use that for setUserID. 

Additionally, I think we should put the max event count check prior to the event list decision so that regardless of the list (queued vs. regular event) we stop event capture

Reply:
I think we should have the check on queued regardless, to ensure we never have a leak unexpectedly ( a poor implementation or something else that uses queued events in the future)